File: CH7/startwithvuejs/src/services/FoodServiceService.ts
import Vue from 'vue'
import { FoodService } from '../entities/FoodService'
import { User } from '../entities/User'

export class FoodServiceService {
  static getById (email: string) {
    return Vue.axios.get<FoodService>(`/foodservices/${email}`)
  }

  static create (foodService: FoodService) {
    return Vue.axios.post<FoodService>(`/foodservices`, foodService)
  }

  static update (foodService: FoodService) {
    return Vue.axios.put<FoodService>(`/foodservices`, foodService)
  }

  static login (user: User) {
    return Vue.axios.post<FoodService>(`/foodservices/login`, user)
  }

  static getByFoodType (foodType:string, page:number, pageSize:number) {
    return Vue.axios.get<Array<FoodService>>(`/foodservices?foodType=${foodType}&page=${page}&pageSize=${pageSize}`)
  }
}

File: CH7/StartWithJEE8/StartWithJEE8-ejb/src/main/java/com/packt/delivery/abstraction/service/foodservice/FoodServiceService.java

package com.packt.delivery.abstraction.service.foodservice;

import com.packt.delivery.abstraction.entity.FoodService;
import com.packt.delivery.abstraction.entity.User;
import java.util.List;
import java.util.Optional;

public interface FoodServiceService {
    Optional<FoodService> login(User user);
    FoodService save(FoodService foodService);
    FoodService update(FoodService foodService);
    FoodService deActivate(String email);
    List<FoodService> getByFoodType(String foodType, Integer page, Integer pageSize);
}
